src/partition/ReceivedBoundingBox.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "Partition.hpp"
+#include "logging/Logger.hpp"
+#include <vector>
+#include "mesh/Vertex.hpp"
+#include "mesh/Mesh.hpp"
+
+
+// Forward delcration to friend the boost test struct
+
+namespace precice {
+namespace partition {
+/**
+ * @brief A partition that is computed from a mesh received from another participant.
+ *
+ * A mesh is received by the master rank and re-partitioned among all slave ranks.
+ * Afterwards necessary distribution data structures are set up.
+ */
+class ReceivedBoundingBox : public Partition
+{
+public:
+
+  /// Defines the typ of geometric filter used
+  enum GeometricFilter {
+    // @brief undefined
+    UNDEFINED,
+    // @brief No geometric filter used (e.g. for RBF mappings)
+    NO_FILTER,
+    // @brief Filter at master and communicate only filtered mesh.
+    FILTER_FIRST,
+    // @brief Broadcast first and filter then
+    BROADCAST_FILTER
```

Comment:
From a language perspective both options are not very consistent.

Random thoughts to start some ideas:
* When do we filter?  `FILTER_FIRST` and `FILTER_LATER`
* What is the order? `FILTER_BROADCAST` and `BROADCAST_FILTER`
* What do we do first? `BROADCAST_FIRST` and `FILTER_FIRST`
* Where do we filter? `FILTER_ON_MASTER` and `FILTER_ON_SLAVES`

Reply:
beyond this PR, on the list for final refactoring work package